tools/DiagramNSX/NsxObjectCapture.ps1

```diff
@@ -44,10 +44,31 @@ If ( (-not $Connection) -and ( -not $Connection.ViConnection.IsConnected ) ) {
 Set-StrictMode -Off
 
 #########################
-$TempDir = "$($env:Temp)\VMware\NSXObjectCapture"
+#Define Windows environment stuff
+if ($psversiontable.PSEdition -ne "Core") {
 $ExportPath = "$([system.Environment]::GetFolderPath('MyDocuments'))\VMware\NSXObjectCapture"
 $ExportFile = "$ExportPath\NSX-ObjectCapture-$($Connection.Server)-$(get-date -format "yyyy_MM_dd_HH_mm_ss").zip"
 
+$TempDir = "$($env:Temp)\VMware\NSXObjectCapture"
+}
+#Overwrite if Core is being used.
+if ($psversiontable.PSEdition -eq "Core") {
+    #Cannot write zip file to export directory on Core
+    $ExportPath = "/tmp/VMware/NSXObjectCaptureOutput"
+    #XML output folder
+    $TempDir = "/tmp/VMware/NSXObjectCapture"
+    #Filename for output
+    $filename = "/NSX-ObjectCapture-$($Connection.Server)-$(get-date -format "yyyy_MM_dd_HH_mm_ss").zip"
+    #FullZIP directory
+    $ExportFile = "$ExportPath/NSX-ObjectCapture-$($Connection.Server)-$(get-date -format "yyyy_MM_dd_HH_mm_ss").zip"
```

Comment:
You've defined the filename as a variable 2 lines up but haven't used it on this line?